### PR TITLE
Steps to implement custom registry id

### PR DESCRIPTION
**Pull request by shhsu**

### CODE REVIEW

File: docs/custom-registry-id.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# How to use a custom regisry ID
+
+Azure docker registries has a typical ID of the format `*-microsoft.azurecr.io`. A customer might like to have a custom registry ID that associate with its own organization. The following is the guide on how to achieve that.
```
Comment (SajayAntony) [inline docs/custom-registry-id.md]: >*-microsoft.azurecr.io` [](start = 56, length = 23)

Please change this to `*.azurecr.io`

File: docs/custom-registry-id.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+## Prerequisites
+
+To setup custom registry ID, the following prerequisites are needed:
+
+* A front end docker host whichs hosts the desired regisry ID. We will run an nginx docker image on it. In this guide we would use the example: `registry.azurecr-test.io`
```
Comment (SajayAntony) [inline docs/custom-registry-id.md]: >whichs [](start = 26, length = 6)

Please try this out - [https://github.com/Microsoft/vscode-spell-check](https://github.com/Microsoft/vscode-spell-check)

Comment (SajayAntony) [inline docs/custom-registry-id.md]: >azurecr-test [](start = 154, length = 12)

rename azurecr-test to  myregistry.contoso.com

File: docs/custom-registry-id.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+1. On `registry.azurecr-test.io`, copy [setup files](custom-id/) into it.
+2. Create a sub-directory named `ssl`, copy `cert.crt`, `private.key`, and `pwd` into this directory. The location of these files can be changed but volume entries in the [docker-compose.yml](custom-id/docker-compose.yml#L11-L13) would then need to be updated to reflect the changes.
+3. Edit [custom_id.env](custom-id/custom_id.env) to point to your front end docker host (E.g. `registry.azurecr-test.io`) and backend Azure Docker Registry (E.g. `peterhsuacr-microsoft.azurecr.io`).
+4. Run `docker-compose up -d` and this should bring up the nginx service that would forward all docker requests from your front end service to the backend Azure Docker Registry.
+
+## Quick verification
+
+A simple way to test the setup is to call `docker login` to quickly confirm that the requests are properly forwarded.
+
+`docker login registry.azurecr-test.io -u myUserName -p 123456`
```
Comment (SajayAntony) [inline docs/custom-registry-id.md]: >myUserName [](start = 42, length = 10)

`docker login registry.contoso.com -u <username> -p <password>`

File: docs/Troubleshooting Guide.md [hunk 4]
```diff
@@ -42,3 +42,6 @@ Please make sure you login before you pull/push repositories
 docker login <yourregistry>.azurecr.io
 ```
 
+# How to use a custom regisry ID
+
+Azure docker registries has a typical ID of the format `*-microsoft.azurecr.io`. A customer might like to have a custom registry ID that associate with its own organization. Follow [this guide](custom-registry-id.md) to achieve that.
```
Comment (SajayAntony) [inline docs/Troubleshooting Guide.md]: >*-microsoft.azurecr.io` [](start = 56, length = 23)

rename to `*.azurecr.io`

**Comment by SajayAntony** [comment]
rename file to custom-domain.md

File: docs/custom-registry-id.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# How to use a custom regisry ID
+
+Azure docker registries has a typical ID of the format `*-microsoft.azurecr.io`. A customer might like to have a custom registry ID that associate with its own organization. The following is the guide on how to achieve that.
+
+## Prerequisites
+
+To setup custom registry ID, the following prerequisites are needed:
```
Comment (SajayAntony) [inline docs/custom-registry-id.md]: >custom registry ID [](start = 9, length = 18)

I think the the concept is "custom domain for azure container registry"

**Comment by SajayAntony** [comment]
:clock1:

File: docs/custom-id/custom_id.env [hunk 6]
```diff
@@ -0,0 +1,2 @@
+FRONTEND_HOST=registry.azurecr-test.io
+BACKEND_HOST=peterhsuacr-microsoft.azurecr.io
```
Comment (SajayAntony) [inline docs/custom-id/custom_id.env]: >peterhsuacr-microsoft.azurecr.io [](start = 13, length = 32)

please rename these to 

registry.contoso.com 
foo-contoso.azurecr.io

**Comment by shhsu** [comment]
Addressed all comments

File: docs/Troubleshooting Guide.md [hunk 7]
```diff
@@ -42,3 +42,6 @@ Please make sure you login before you pull/push repositories
 docker login <yourregistry>.azurecr.io
 ```
 
+# How to use a custom domain for azure container registry
+
+Azure docker registries has a typical ID of the format `*.azurecr.io`. A customer might like to have a custom domain that associate with its own organization. Follow [this guide](custom-domain.md) to achieve that.
```
Comment (SajayAntony) [inline docs/Troubleshooting Guide.md]: >a typical ID [](start = 28, length = 12)

Please rename "typical ID" to login URL

**Pull request merged**